numpy/core/tests/test_numeric.py

```diff
@@ -140,6 +140,36 @@ def test_round(self):
         tgt = [1.6, 72.5, 6.4, 3.2]
         assert_equal(np.around(arr, decimals=1), tgt)
 
+        s = np.float64(1.)
+        assert_(isinstance(round(s), int))
+        assert_(isinstance(round(s, None), int))
+        assert_(isinstance(round(s, ndigits=None), int))
+        assert_(isinstance(s.round(), np.float64))
+        assert_equal(s.round(), 1.)
+        assert_equal(round(s), 1)
+        assert_equal(round(s, None), 1)
+        assert_equal(round(s, ndigits=None), 1)
+
+        s = np.uint64(2**64-1)
+        assert_equal(round(s), 2**64-1)
+
+        f = np.float64(5.1 * 10**73)
+        assert_(isinstance(round(f, -73), np.float64))
+        assert_allclose(round(f, -73), 5.0 * 10**73)
+        assert_(isinstance(round(f, ndigits=-73), np.float64))
+        assert_allclose(round(f, ndigits=-73), 5.0 * 10**73)
+
+        i = np.int64(501)
+        assert_(isinstance(round(i, -2), np.int64))
+        assert_allclose(round(i, -2), 500)
+        assert_(isinstance(round(i, ndigits=-2), np.int64))
+        assert_allclose(round(i, ndigits=-2), 500)
+
+    @pytest.mark.xfail
+    def test_round_py_consistency(self):
+        f = 5.1 * 10**73
+        assert_equal(round(np.float64(f), -73), round(f, -73))
```

Comment:
Interesting, any idea why we are getting a different result?

Reply:
None, unfortunately. This is a bug in `np.round`, possibly a floating-point error.